chain/index/ddls.go

```diff
@@ -0,0 +1,98 @@
+package index
+
+import "database/sql"
+
+// TODO PRAGMA foreign_keys = ON;
+// TODO: Allow for msg_cid <> eth_tx_hash for unconfirmed mpool messages
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tipset_message (
+		message_id INTEGER PRIMARY KEY,
+		tipset_key_cid BLOB NOT NULL,
+		height INTEGER NOT NULL,
+		reverted INTEGER NOT NULL,
+		message_cid BLOB NOT NULL,
+		message_index INTEGER NOT NULL,
+		events_processed INTEGER NOT NULL,
```

Comment:
what's the purpose of this field btw?

Reply:
@rvagg Eventually, want the event read API to be differentiate between "events processed and no events" vs "yet to process events". This is a simple denormalised field that will help with that.

Reply:
For a tipset T, events in T will be processed when we index a tipset U such that Parent(U)=T.

Reply:
but do we use this anywhere? I'm not seeing it, what's the plan?

Reply:
@rvagg Yeah lemme add a "ReadEventsAPI" to show it'll be used and wire it into the `EventFilterManager`.

Reply:
This PR was to show the general direction I am taking. Next one will be code complete.